### DISCUSSION TITLE
New speaker encoder

### DISCUSSION THREAD

**Discussion by loganhart02** [coqui-ai/TTS#447] [General Q&A]
Hey guys! I've been working on trying to improve the multi-speaker models and I got access to a spotify podcast dataset(It is 2TB of several podcast). I haven't really got dirty with messing around with it yet because of the startup I'm founding so I'm not sure how many speakers there really are but I know that these are full length podcast and from what I've listened to they are of good quality too. Do you guys think that if we trained a new speaker encoder that it would improve on the current one to get a better multi-speaker TTS system? Wanted to ask before I dived in deep into trying to preprocess this massive dataset.

**Comment by erogol** [comment]
I think it'd be definitely valuable for some use cases. Current speaker encoder is trained by @mueller91, so he might comment on it better.

**Comment by mueller91** [comment]
For a good speaker encoder, a high number of different speakers in the dataset is critical. How many speakers does your dataset contain?

**Reply by loganhart02** [reply]
I haven't checked exactly how many speakers there are yet(I'm still converting all the audio to .wav format) but I do know there are 100,000 podcasts so I would assume the number is fairly high. It's taking a while for all the files to convert but once they do I'll have a better idea.

**Reply by mueller91** [reply]
if i remember correctly, i trained on about 20k speakers. I used VoxCeleb1+2, Mozilla CommonVoice and LibriTTS, but only those speakers with more than 10 utterances of length.
So if you have 100k speakers, this would be quite an improvement. But i'm doubtful about the number of speakers in your dataset, usually a single podcaster creates multiple podcasts... so it may be much less than 100k individual speakers. And you'd have to label all audio files with a unique speaker ID.

can you provide more information on your data?

**Reply by loganhart02** [reply]
I'm currently at work but I'll be able to have much more information on speaker information tonight/tomorrow morning. here is the link to the actual dataset as well. It doesn't really give any information on number of speakers but it provides a lot of information on the dataset as a whole. I assumed all episodes wouldn't have a different speakers but even if it's say 10 episodes to a speaker that 10k extra speakers to use on top of the 20k for all the other datasets. another thing I was curious about is the dataset has 50000 hours of speech. now assuming there is at the very least 10k speakers. could we use this to train a new multi-speaker tacotron2? 
link to dataset: https://podcastsdataset.byspotify.com/

**Reply by loganhart02** [reply]
sorry for the late follow up, finals had me down bad. after exploring a small portion of the dataset there is actually more than one speaker per podcast episode(usually Two) so intuitively it is probably correct to assume there are close to 100k speakers in the entire set although the amount of samples per speaker is very skewed. usually there is one speaker(the one who own the podcast) who is in all there episodes(around 10) so they have hours of samples. but the guest isn't in every episode. Knowing this what would you guys suggest be the minimum number of samples to qualify a speaker to use in the dataset(in terms of minutes/hours) and how many speakers would be the minimum to have to use this for the speaker encoder training and multi speaker TTS training?

**Reply by Edresson** [reply]
I think that using this dataset it is a little difficult to separate the speakers. The separation of the speakers is very important for the loss functions. I would suggest using datasets that already have speaker separation as common voice.

I am training a new speaker encoder. See PR https://github.com/coqui-ai/TTS/pull/508
I'm training with 53k speakers (Common voice all languages + voxceleb1 + voxceleb2)  

The model has been training for 3 days and the results are already better than our current speaker encoder :).

**Reply by astricks** [reply]
@Edresson is the speaker encoder language-independant? For some reason, I was under the impression it is constrained to the English language.

**Reply by Edresson** [reply]
@astricks  In theory the speaker encoder should not work with similar performance in any language. During training our speaker encoder saw much more audio in English without any balance, but it should work. Training with multiple languages can help (especially given the number of extra speakers and new phonemes).

**Reply by astricks** [reply]
Interesting.. I’ll try it with Hindi and see how that works. Looks like you trained with all of Mozilla common language so the encoder has seen some Hindi speakers.

**Reply by astricks** [reply]
I was also wondering - would I be able to effectively use the speaker encoder to generate speaker embeddings on my dataset even if I am using a higher sample rate, say, 48k?
